spec/integration/multiple_threadpools_spec.rb

```diff
@@ -0,0 +1,29 @@
+class DifferentThreadpoolsSubscriber < ActionSubscriber::Base
+  def one
+    $messages << payload
+  end
+
+  def two
+    $messages << payload
+  end
+end
+
+describe "Separate Threadpools for Different Message", :integration => true do
+  let(:draw_routes) do
+    low_priority_threadpool = ::ActionSubscriber::Threadpool.new_pool(:low_priority, 1)
+    ::ActionSubscriber.draw_routes do
+      route DifferentThreadpoolsSubscriber, :one
+      route DifferentThreadpoolsSubscriber, :two, :threadpool => low_priority_threadpool
+    end
+  end
+
+  it "processes messages in separate threadpools based on the routes" do
+    ::ActionSubscriber.auto_subscribe!
+    ::ActionSubscriber::Publisher.publish("different_threadpools.one", "ONE", "events")
+    ::ActionSubscriber::Publisher.publish("different_threadpools.two", "TWO", "events")
+
+    verify_expectation_within(2.0) do
+      expect($messages).to eq(Set.new(["ONE","TWO"]))
+    end
+  end
+end
```

Comment:
I tried writing the test that way a few times, but because the timing of publishing the message to rabbitmq and getting it back from rabbitmq is asynchronous and varies a lot between running locally vs running in travis I just couldn't get that test to run consistently.

That's why I changed it to just be a smoke test.